dali/pipeline/operators/util/pad_sparse_batch.cu

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map>
+#include "dali/pipeline/operators/util/pad_sparse_batch.h"
+#include "dali/core/static_switch.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+namespace detail {
+
+template <typename Type>
+void RunHelper(TensorList<GPUBackend> &output,
+               const TensorList<GPUBackend> &input,
+               Type padding_val,
+               int axis,
+               cudaStream_t stream,
+               kernels::ScratchpadAllocator &scratch_alloc) {
+  std::size_t number_of_dims = input.tensor_shape(0).size();
+  VALUE_SWITCH(number_of_dims, NumDims, (2, 3, 4), (
+    kernels::PadGPU<Type, NumDims> kernel;
+    kernels::KernelContext ctx;
+    ctx.gpu.stream = stream;
+    auto in_view = view<const Type, NumDims>(input);
+
+    kernels::KernelRequirements req = kernel.Setup(ctx, in_view, axis);
+
+    output.set_type(TypeInfo::Create<Type>());
+    output.Resize(req.output_shapes[0]);
+
+    scratch_alloc.Reserve(req.scratch_sizes);
+    auto scratchpad = scratch_alloc.GetScratchpad();
+    ctx.scratchpad = &scratchpad;
+
+    auto out_view = view<Type, NumDims>(output);
+    // TODO(spanev): add per sample padding val
+    kernel.Run(ctx, out_view, in_view, padding_val);
+  ),  // NOLINT
+  (
+    DALI_FAIL("Not supported number of dimensions: " + std::to_string(number_of_dims));
+  ));  // NOLINT
+}
+
+}  // namespace detail
+
+DALI_SCHEMA(PadSparseBatch)
+    .DocStr(R"code(Pads a batch of non-uniform shape element of with `padding_value`.
```

Comment:
Pads up to what?
I would add an option to provide the external size of the axis and all samples, in all batches are padded up to this dimension (sizes in all batches are the same), or just let if infer from the sample size in the batch (samples sizes in batch are same, but batch to batch can vary).

Reply:
I will rework this docstring and add some examples